src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsHostBuilderExtensions.cs

```diff
@@ -58,5 +134,22 @@ public static IHostBuilder ConfigureWebJobs(this IHostBuilder builder, Action<Ho
 
             return builder;
         }
+
+        private static IConfigurationBuilder TryAddDefaultConfigurationSources(this IConfigurationBuilder config)
+        {
+            if (!config.Sources.OfType<JsonConfigurationSource>().Any(p => string.Compare(p.Path, "appsettings.json", StringComparison.OrdinalIgnoreCase) == 0))
```

Comment:
nit: consider using `string.Equals` instead of `string.Compare` for better clarity (since you're actually testing the values for equality, not comparing them for ordering)